### PR TITLE
Support Canon iP72*0 and other printers - Work in progress, help needed

### PR DESCRIPTION
**Pull request by turion**
###### Motivation for this change

Canon iP72*0 does not work in nixos currently.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

# Questions

I'm not a C dev and I have no experience with `autoconf`. I simply copied `pkgs/misc/cups/drivers/cnijfilter_4_00/default.nix`, downloaded the source from the Canon website (http://gdlp01.c-wss.com/gds/3/0100004693/01/cnijfilter-source-3.80-1.tar.gz) and tried to fiddle around to get as far as I could.

* Did I do most things right so far?
* How can I improve my code?
* Right now, it doesn't build. the build error is:
```
gcc  -O2 -L../../../com/libs_bin64 -L../../../lgmon/src  -o cnijnpr cnijnpr.o -lcnnet -lbscc2sts -ldl 
/nix/store/sq2b0dqlq243mqn4ql5h36xmpplyy20k-binutils-2.31.1/bin/ld: cannot find -lbscc2sts
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:337: cnijnpr] Error 1
make[2]: Leaving directory '/build/source/cngpijmon/cnijnpr/cnijnpr'
make[1]: *** [Makefile:353: all-recursive] Error 1
make[1]: Leaving directory '/build/source/cngpijmon/cnijnpr'
make: *** [Makefile:294: all] Error 2
```
What can I do here? It seems the file `bscc2sts.c` is in the `lgmon/src` directory, but still the linker doesn't find it.
* @chpatrick @jerith666 you added similar wrappers before, can you give some guidance?

### CODE REVIEW

File: pkgs/misc/cups/drivers/cnijfilter_3_80/default.nix [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+     pkgs/development/tools/misc/patchelf/setup-hook.sh) is run on
+     them, it undoes the --set-rpath.  this prevents that. */
+  dontPatchELF = true;
+
+  meta = with lib; {
+    description = "Canon InkJet printer drivers for the ${models} series.";
+    homepage = https://www.canon.co.uk/support/consumer_products/products/printers/inkjet/pixma_ip_series/pixma_ip7250.html?type=drivers&language=&os=linux;
+    license = licenses.unfree;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ turion ];
```
Comment (cole-h) [inline pkgs/misc/cups/drivers/cnijfilter_3_80/default.nix:157]: I can't help with the actual implementation or testing, but I can help you make ofborg happy: you need to add yourself as a maintainer for this to be valid. See [`maintainer-list.nix`](https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix) for more information, and my maintainers commit for an example: https://github.com/NixOS/nixpkgs/commit/ecf79a07e1cf1d895b058d33def554ae7c35283d.

**Comment by jerith666** [comment]
I don't have any easy ideas, sorry.  :(  I know this was quite a pain to get working correctly, I wish you luck!

File: pkgs/misc/cups/drivers/cnijfilter_3_80/default.nix (hunk 1 above)
Comment (turion) [reply pkgs/misc/cups/drivers/cnijfilter_3_80/default.nix:157]: Thanks!

**Comment by turion** [comment]
Oh it seems there is a more complete PR here: https://github.com/NixOS/nixpkgs/pull/65469

**Comment by turion** [comment]
There is a more complete PR, and I don't have access to this kind of printer anymore.

**Pull request closed without merging**